opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py

```diff
@@ -343,8 +321,29 @@ def __init__(
         export_interval_millis: Optional[float] = None,
         export_timeout_millis: Optional[float] = None,
     ) -> None:
+        _instrument_class_temporality = {}
+        # Exporter-specific logic for otlp
+        if exporter.__class__.__name__.lower().__contains__("otlp"):
```

Comment:
I thought a lot a bout what can we do here. I understand that this is a very difficult situation because the setting of `_instrument_class_temporality` happens in the `MetricReader` but in this case that setting depends on the specific exporter being an OTLP exporter.

I can't figure out a clean solution out of this situation, but we can be "less hacky" if instead of checking if `"otlp"` is in the name of the class, create a new class somewhere in the SDK, a purpose-specific `OTLPMetricsExporterMixin`. Make the OTLP metrics exporter (and nothing else) inherit from it, then change this check for an `isinstance`.

Reply:
Actually, now that I think about it, the approach I suggested above is also hacky and it increases the public API. I think the approach in this PR is fine, then, it is just not possible to not be hacky under these circumstances.

Only one thing, please add a comment that mentions this is a hack, since anyone looking at this code later would be wondering why we implemented this in this way :+1:

Reply:
Added a comment.